packages/backend-dynamic-feature-service/src/loader/CommonJSModuleLoader.ts

```diff
@@ -44,11 +62,80 @@ export class CommonJSModuleLoader implements ModuleLoader {
           dynamicNodeModulesPaths.some(p => nodeModulePath.startsWith(p))
         );
       });
-      this.logger.debug(
+      this.options.logger.debug(
         `Overriding node_modules search path for dynamic plugin ${from} to: ${filtered}`,
       );
       return filtered;
     };
+
+    const oldResolveFileName = this.module._resolveFilename;
+    this.module._resolveFilename = (
```

Comment:
Probably best if we could avoid this tbh. I wouldn't necessarily be against modifying the behavior of `resolvePackagePath` tbh, depending on what that'd look like. Especially if we can make it work with a future bundled layout for backend packages where they might all be mixed under a single directory

Reply:
I agree that having the `resolvePackagePath` be more flexible would be great.

However, without this fix, the `CommonJSModuleLoader` default implementation of the dynamic backend plugin `ModuleLoader` is nearly useless, since it just doesn't support database-enabled dynamic plugins.
That's why we've been using a copied & fixed version of the CommonJSModuleLoader since 7 months on the Janus side (see the note above).

And it seems to me that, even when/if the `resolvePackagePath` utility is enhanced, this fix would still be necessary to support any dynamic backend plugin using the old version of `resolvePackagePath`.

Reply:
As mentioned in the framework SIG call, I added a comment just above the `_resolveFileName` overriding, in order to provide context, and the fact that it might not be necessary anymore, when `resolvePackagePath` is made more flexible (or an equivalent, more flexible utility is provided).